src/live-programming.rst

```diff
@@ -36,8 +36,150 @@ Direct Manipulation Programming
+  Users can import or create images and sounds within the editor.
+  Programming is done by dragging and snapping together colorful command blocks to control 2D graphical objects called sprites moving on a background called the stage.
+  This paper also describes aspects of Scratch and the language design that make it easier for young people to explore, express themselves, and learn.
+
+The Blueprints Visual Scripting system :cite:`blueprints` in Unreal Engine is a gameplay scripting system that uses the concept of dataflow programming to compose game elements from within the Unreal Editor.
+Users can use simple drag-and-drop operations to draw connections between nodes on the interface without writing code.
+The system is used to define object-oriented (OO) classes and objects in the engine.
+Specifically, Blueprints can handle extending classes, storing and modifying default properties, and managing components instancing for classes.
+
+
+Game Description Languages
+----------------------------------
+Game description languages seek to express components expected in the state of a game, and the rules that induce transitions, resulting in a state-action space.
+Such languages have the potential of enabling automatic game generation and offer opportunities to formalize the knowledge involved in game design and test game design theories.
+Several attempts have been made in the past to model aspects of games and to encode game mechanics for analysis.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'ebner2013towards'
+
+  This paper discusses the key requirements and challenges in constructing a new Video Game Description Language (VGDL).
+  It proposed an initial design of the semantics of the language and the components required to define a given game.
+  The core components required in order to represent a video game include map, objects, player definitions, avatars, physics, events, and rules.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'martens2015ceptre'
+
+  This paper introduces Ceptre, a rule specification language to enable rapid prototyping for experimental game mechanics.
+  Ceptre presents a correspondence between *gameplay* and *proof search* in linear logic.
+  This methodology is proposed to help game designers and researchers in designing, analyzing, and debugging generative, multi-agent gameplay.
+
+Game Development in Education
+-------------------------------------------------
+Researchers have developed game development environments to broaden interest in computing amongst students in K-12 and university settings.
+
+.. container:: bib-item
+
+  .. bibliography:: game-development.bib
+    :filter: key == 'maloney2008programming'
+
+  This paper reports on the Scratch programming experiences of urban youth from after-school programs over an 18-month period.
```

Comment:
again not clear whether video games were a specific topic of interest in this paper